oauth2client/client.py

```diff
@@ -984,10 +982,10 @@ def _detect_gce_environment():
     #       could lead to false negatives in the event that we are on GCE, but
     #       the metadata resolution was particularly slow. The latter case is
     #       "unlikely".
-    http = transport.get_http_object(timeout=GCE_METADATA_TIMEOUT)
     try:
-        response = transport.request(
-            http, _GCE_METADATA_URI, headers=_GCE_HEADERS)
+        response = _transport.request(
+            None, _GCE_METADATA_URI, headers=_GCE_HEADERS,
```

Comment:
See the logic in `_transport.request`, `http` can now explicitly be `None` and the preferred transport will be used. This is reflected in the docstring.